### PR TITLE
Remove code that supported optional numba installation

### PR DESCRIPTION
**Pull request by mmcky**
This PR removes code that was written to support environments that did't have ``numba`` installed. Now that ``numba`` is a strict dependency of the package - this code can now be removed. This will simplify how code is written etc. 

**Updates:**
  1. ``markov/gth_solve`` should we remove the following given support for Python 3.5+:

```python
+#-TODO: Remove as we no longer support Python <3.5?
+try:
+    xrange
+except:  # python3
+    xrange = range
```

  2. ``gth_solve`` so as not to change the code to much, I have added a ``use_jit`` header to the function signature to allow one to optionally use numpy. @oyamad Is this ok?

**Deprecations:**
  1. The ``numba_installed`` package wide variable is **no** longer available.

### CODE REVIEW

**Comment by mmcky** [comment]
In #246 It was resolved to remove the python2.7 range statements.

**Comment by mmcky** [comment]
I will leave this open for a day - in case there are any comments on the final diff.

File: quantecon/lss.py [hunk 1]
```diff
@@ -9,10 +9,10 @@
 import numpy as np
 from numpy.random import multivariate_normal
 from scipy.linalg import solve
+from numba import jit
 
-#-Check if Numba is Available-#
-from .util import numba_installed, jit
-
+#TODO: Should this be specified with (nopython=True?)
```
Comment (oyamad) [inline quantecon/lss.py]: `@jit(nopython=True)` will work if the line `A = np.asarray(A)` is removed (assuming that the input `A` is an numpy ndarray). (But I suspect we wouldn't get much speedup.)

File: quantecon/random/utilities.py [hunk 2]
```diff
@@ -75,6 +77,8 @@ def _diff(r, out):
         out[i, n] = 1 - r[i, n-1]
 
 
+#TODO: Should this use (nopython=True) flag?
```
Comment (oyamad) [inline quantecon/random/utilities.py]: `nopython=True` will not work, due to `random_state` at least.

File: quantecon/markov/random.py [hunk 3]
```diff
@@ -218,7 +219,8 @@ def random_discrete_dp(num_states, num_actions, beta=None,
     ddp = DiscreteDP(R, Q, beta, s_indices, a_indices)
     return ddp
 
-
+#TODO: Should this use (nopython=True) flag?
```
Comment (oyamad) [inline quantecon/markov/random.py]: `nopython=True` should work if `dtype=int` is replaced with `dtype=np.int64`. (Again I suspect we won't have much speed up.)

Comment (oyamad) [inline quantecon/markov/random.py]: In a (near) future PR, I would like to make this function `_sa_indices` public. One possibility is to move it to a new file `markov/utilities.py`. Another is to put it to `gridtools.py`, with an appropriate name?

File: quantecon/lss.py (hunk 1 above)
Comment (mmcky) [inline quantecon/lss.py]: Thanks @oyamad Agree. Let's leave this as is.

File: quantecon/markov/random.py (hunk 3 above)
Comment (mmcky) [inline quantecon/markov/random.py]: Thanks @oyamad.  Let's accept this PR and then we can update this function.

**Comment by mmcky** [comment]
Thanks for your comments @oyamad. I will now remove the ``#TODO`` statements.

**Pull request merged**